stablehlo/reference/Ops.cpp

```diff
@@ -68,6 +81,49 @@ Tensor eval(FloorOp op, const Tensor &operand) {
   return result;
 }
 
+Tensor eval(IotaOp op) {
+  Tensor result(op.getType());
+  Type elType = result.getType().getElementType();
+  if (!isSupportedIntegerType(elType) && !isSupportedFloatType(elType) &&
+      !isSupportedComplexType(elType))
+    report_fatal_error(invalidArgument("Unsupported element type: %s",
+                                       debugString(elType).c_str()));
```

Comment:
It looks like this check duplicates the checks that are already done in the loop. E.g. if this was happening inside the loop, the only thing needed would be `else report_fatal_error(...)`. Any specific reason to prefer the current arrangement?

Reply:
It's a combination of early termination preference and an attempt to remove some clutter in the loop. With your suggestion, below, I'll move this in the loop.